CASMPET-5640 Corrected VCS Password Update instructions (release/1.0) (#1657)
Cray-HPE/docs-csm: operations/configuration_management/Version_Control_Service_VCS.md

                  value: my_secret_password
              - type: static
                args:
                  name: vcs_username
                  value: crayvcs
    ```

1. Upload the modified `customizations.yaml` file to Kubernetes.

   ```bash
   ncn# kubectl delete secret -n loftsman site-init
   ncn# kubectl create secret -n loftsman generic site-init --from-file=customizations.yaml
   ```

1. Encrypt the values after changing the `customizations.yaml` file.

    ```bash
    ncn# ./secrets-seed-customizations.sh customizations.yaml
    ```

   If the above command complains that it cannot find `certs/sealed_secrets.crt`, then you can run the following commands to create it:

    ```bash
    ncn# mkdir -p ../certs &&
         ./bin/linux/kubeseal --controller-name sealed-secrets --fetch-cert > ../certs/sealed_secrets.crt
    ```

1. Create a local copy of the `sysmgmt.yaml` file.

    ```bash
    ncn# kubectl get cm -n loftsman loftsman-sysmgmt -o jsonpath='{.data.manifest\.yaml}'  > sysmgmt.yaml
    ```

1. Edit the `sysmgmt.yaml` to only include the `gitea` chart and all its current data.

   Example:

    ```yaml
    apiVersion: manifests/v1beta1
      metadata:

>         name: sysmgmt

Author: @kburns-hpe Made changes as you suggested. Can you please review? Thanks

>         name: sysmgmt

Reviewer: looks good.

**WARNING:** These three sources of credentials are not synced by any mechanism. Changing the default password requires that is it changed in all three places. Changing only one may result in difficulty determining the password at a later date, or may result in losing access to VCS altogether.

To change the password in the `vcs-user-credentials` Kubernetes secret, use the following procedure:

> 1. Log in to Keycloak with the default admin credentials.

Reviewer 2: ```suggestion
1. Log in to Keycloak with the default `admin` credentials.
```

> 1. Log in to Keycloak with the default admin credentials.

Reviewer 2: ```suggestion
1. Log in to Keycloak with the default `admin` credentials.
```

>    Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing SYSTEM\_DOMAIN\_NAME with the actual NCN's DNS name.

Reviewer 2: ```suggestion
   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing `SYSTEM_DOMAIN_NAME` with the actual NCN's DNS name.
```
```suggestion
   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing SYSTEM\_DOMAIN\_NAME with the actual NCN's DNS name.
```

>    Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing SYSTEM\_DOMAIN\_NAME with the actual NCN's DNS name.

Reviewer 2: ```suggestion
   Point a browser at `https://auth.SYSTEM_DOMAIN_NAME/keycloak/admin`, replacing `SYSTEM_DOMAIN_NAME` with the actual NCN's DNS name.
```

>    The following is an example URL for a system: 'https://auth.cmn.system1.us.cray.com/keycloak/admin`

Reviewer 2: ```suggestion
   The following is an example URL for a system: `https://auth.cmn.system1.us.cray.com/keycloak/admin`
```

>    The following is an example URL for a system: 'https://auth.cmn.system1.us.cray.com/keycloak/admin`

Reviewer 2: ```suggestion
   The following is an example URL for a system: `https://auth.cmn.system1.us.cray.com/keycloak/admin`
```
```suggestion
   The following is an example URL for a system: 'https://auth.cmn.system1.us.cray.com/keycloak/admin`
```

>    Use the following admin login credentials:

Reviewer 2: ```suggestion
   Use the following `admin` login credentials:
```

>    Use the following admin login credentials:

Reviewer 2: ```suggestion
   Use the following `admin` login credentials:
```

   - Username: `admin`
   - The password can be obtained with the following command:

>     ```bash

Reviewer 2: Indent this code block under the above bullet

    ncn# kubectl get secret -n services keycloak-master-admin-auth \
                 --template={{.data.password}} | base64 --decode
    ```

1. Ensure the selected Realm is `Shasta` from the top-left dropdown in the left sidebar.

1. From the left sidebar, under the `Manage` section, select `Users`.

1. In the `Search...` textbox, type in `crayvcs` and click the search icon.

1. In the filtered table below, click on the ID for the row that shows `crayvcs` in the `Username` column. 

1. Go to the `Credentials` tab and change the password.

   Enter the new password in the `Reset Password` form. Ensure `Temporary` is switched **off**. Click on `Reset Password` button.

> 1. Now log in to Gitea with the default admin credentials.

Reviewer 2: ```suggestion
1. Log in to Gitea with the default `admin` credentials.
```

> 1. Now log in to Gitea with the default admin credentials.

Reviewer 2: ```suggestion
1. Log in to Gitea with the default `admin` credentials.
```

   Point the browser at `https://vcs.SHASTA_CLUSTER_DNS_NAME/vcs/user/settings/account`.

   If presented with Keycloak login, use `crayvcs` as the username and the new VCS password. Wait to be redirected to the Gitea login page before continuing to the next step.

1. Use the following Gitea login credentials:

   - Username: `crayvcs`
   - The old VCS password, which can be obtained with the following command:

>     ```bash

Reviewer 2: Indent this code block under the above bullet

>     ```bash

Reviewer 2: Indent this code block under the above bullet

    ncn# kubectl get secret -n services vcs-user-credentials \
            --template={{.data.vcs_password}} | base64 --decode
    ```

> 1. Enter the existing password (from previous step), new password and confirmation, and then click `Update Password`.

Reviewer 2: ```suggestion
1. Enter the existing password (from previous step), new password, and confirmation, and then click `Update Password`.
```
```suggestion
1. Enter the existing password (from previous step), new password and confirmation, and then click `Update Password`.
```

> 1. Enter the existing password (from previous step), new password and confirmation, and then click `Update Password`.

Reviewer 2: ```suggestion
1. Enter the existing password (from previous step), new password, and confirmation, and then click `Update Password`.
```

1. Now SSH into `ncn-w001` or `ncn-m001`.
1. Run `git clone https://github.com/Cray-HPE/csm.git`.

> 1. Copy the directory `vendor/stash.us.cray.com/scm/shasta-cfg/stable/utils` to your desired working directory, and run the following commands from that working directory.

Reviewer 2: ```suggestion
1. Copy the directory `vendor/stash.us.cray.com/scm/shasta-cfg/stable/utils` to the desired working directory.
1. Change directories to be in the working directory set in the previous step.
```

> 1. Copy the directory `vendor/stash.us.cray.com/scm/shasta-cfg/stable/utils` to your desired working directory, and run the following commands from that working directory.

Reviewer 2: ```suggestion
1. Copy the directory `vendor/stash.us.cray.com/scm/shasta-cfg/stable/utils` to the desired working directory.
1. Change directories to be in the working directory set in the previous step.
```

   In the `customizations.yaml` file, set the values for the `gitea` keys in the `spec.kubernetes.sealed_secrets` field. The value in the data element where the name is `password` needs to be changed to the new Gitea password. The section below will replace the existing sealed secret data in the `customizations.yaml` file.

   For example:
    ```yaml
          gitea:
            generate:
              name: vcs-user-credentials
              data:
              - type: static
                args:
                  name: vcs_password
                  value: my_secret_password
              - type: static
                args:
                  name: vcs_username
                  value: crayvcs
    ```

1. Upload the modified `customizations.yaml` file to Kubernetes.

   ```bash
   ncn# kubectl delete secret -n loftsman site-init
   ncn# kubectl create secret -n loftsman generic site-init --from-file=customizations.yaml
   ```

1. Encrypt the values after changing the `customizations.yaml` file.

    ```bash
    ncn# ./secrets-seed-customizations.sh customizations.yaml
    ```

>    If the above command complains that it cannot find `certs/sealed_secrets.crt` then you can run the following commands to create it:

Reviewer 2: ```suggestion
   If the above command complains that it cannot find `certs/sealed_secrets.crt`, then you can run the following commands to create it:
```
```suggestion
   If the above command complains that it cannot find `certs/sealed_secrets.crt` then you can run the following commands to create it:
```

>    If the above command complains that it cannot find `certs/sealed_secrets.crt` then you can run the following commands to create it:

Reviewer 2: ```suggestion
   If the above command complains that it cannot find `certs/sealed_secrets.crt`, then you can run the following commands to create it:
```
